### ISSUE TITLE
Filtering by last year providing no results

### ISSUE THREAD

**Issue by ghost** [metabase/metabase#3403]
I have a table in a sqlite database with a 'created_date' column. 

When trying to get raw results from THIS year, the generated sql filter looks like this:

``` sql
WHERE CAST(strftime('%Y', "table"."created_date") AS integer) = CAST(strftime('%Y', datetime('now')) AS integer)
```

And this correctly finds all rows created in 2016.

However, when I try to filter to return rows from LAST year, it incorrectly returns 0 results. Looking into the generated SQL, I see the following filter:

``` sql
WHERE CAST(strftime('%Y', "table"."created_date") AS integer) = CAST(strftime('%Y', datetime(CAST(strftime('%Y', 'now') AS integer), '-1 years')) AS integer)
```

The right hand side of this filter:

``` sql
CAST(strftime('%Y', datetime(CAST(strftime('%Y', 'now') AS integer), '-1 years')) AS integer)
```

actually evaluates to be `-470`, and not the expected value of 2015.

For what it's worth, timestamps are stored in this format: `2016-10-26 00:00:00`

**Comment by camsaul** [comment]
@paulinder what SQL would you expect it to produce to give you the correct results?

**Comment by paulinder** [comment]
Hi Cam,

### This is what I first did
The generated SQL filter has this:
```sql
WHERE CAST(strftime('%Y', "table"."created_date") AS integer) = CAST(strftime('%Y', datetime(CAST(strftime('%Y', 'now') AS integer), '-1 years')) AS integer)
```
and the right side evaluates to `-470`, providing no results.

I can change this to my own:
```sql
WHERE CAST(strftime('%Y', "table"."created_date") AS integer) = CAST(strftime('%Y', 'now') AS integer) - 1
```
where the right side evaluates to `2015`, providing the correct results. Just to be explicitly clear, I'm simply subtracting 1 from the integer representation of the current year...


### A better solution
I found that filtering by 'last week' and 'last month' already work, so looking at those filters I can see they look like this:
```sql
WHERE date("table"."created_date", 'start of month') = date(datetime(date('now', 'start of month'), '-1 months'), 'start of month')
```

So, I simply changed month -> year:
```sql
WHERE date("table"."created_date", 'start of year') = date(datetime(date('now', 'start of year'), '-1 years'), 'start of year')
```
and that also works.

**Comment by salsakran** [comment]
@camsaul I believe that the change to using date ranges instead of date extracts for datetime filtering rewrites this entire section, and this bug should be closed?

**Comment by flamber** [comment]
On 0.35.4 with SQLite it produces the following - not sure when it was fixed, but at least from 0.34:
```
WHERE date("table"."created_date", 'start of year') = date(datetime(date(datetime('now'), 'start of year'), '-1 years'), 'start of year')
```
